azerothcore/azerothcore-wotlk
# chore(SmartAI): handle unused SmartActions

##### CHANGES PROPOSED:

1) Emit error when there is a SmartAction in the DB that is not supposed to be used in version 3.3.5a:

> SmartAIMgr: EntryOrGuid %d using event(%u) has an action type that is not supported on 3.3.5a (%u), skipped.

(this is useful to warn users that try to import stuff from other expansions).

```
- SMART_ACTION_RESERVED_16                        = 16
- SMART_ACTION_PLAY_ANIMKIT                       = 128
- SMART_ACTION_SCENE_PLAY                         = 129
- SMART_ACTION_SCENE_CANCEL                       = 130
```

2) Emit error when there is a SmartAction in the DB that is not yet supported in AzerothCore:
>SmartAIMgr: EntryOrGuid %d using event(%u) has an action type that is not yet supported on AzerothCore (%u), skipped.

(this is useful to warn users that try to import stuff from TC)

Currently, those (not yet supported) actions are:
```
- SMART_ACTION_RANDOM_SOUND                       = 115
- SMART_ACTION_SET_CORPSE_DELAY                   = 116
- SMART_ACTION_DISABLE_EVADE                      = 117
- SMART_ACTION_GO_SET_GO_STATE                    = 118
- SMART_ACTION_SET_CAN_FLY                        = 119
- SMART_ACTION_REMOVE_AURAS_BY_TYPE               = 120
- SMART_ACTION_REMOVE_MOVEMENT                    = 127
- SMART_ACTION_SPAWN_SPAWNGROUP                   = 131
- SMART_ACTION_DESPAWN_SPAWNGROUP                 = 132
- SMART_ACTION_RESPAWN_BY_SPAWNID                 = 133
```


###### ISSUES ADDRESSED:

- Updates https://github.com/azerothcore/azerothcore-wotlk/issues/1326


##### TESTS PERFORMED:

Build, runs.


##### HOW TO TEST THE CHANGES:

- Insert a `smart_scripts` entry that has an invalid smart action (`action_type`) and make sure that when the `worldserver` runs it correctly gives the startup DB error
- Ideally try one error for each of the 2 categories (one non-3.3.5, one non-yet-supported-in-AC)

##### Target branch(es):

Master
